src/documents/CoreAPI.md

```diff
@@ -0,0 +1,635 @@
+list member to be transformed, and `index` is its index. Each function will be
+applied in turn to each member, which will be replaced by the return value from the function.
+* Returns: `{Array|Object}` The finally transformed structure, where each member has been replaced by the original member acted on by the function or functions.
+
+### fluid.accumulate(list, fn, initial)
+
+Scan through a list of objects, "accumulating" a value over them
+(may be a straightforward "sum" or some other chained computation). "accumulate" is the name derived
+from the C++ STL, other names for this algorithm are "reduce" or "[fold](https://en.wikipedia.org/wiki/Fold_%28higher-order_function%29)".
+
+* `list {Array}` The list of objects to be accumulated over.
+* `fn {Function: (object {Any}, total {Any-Total}, index {Integer}) → Any-Total}` An "accumulation function" accepting the signature `(object, total, index)` where
+`object` is the list member, `total` is the "running total" object (which is the return value from the previous invocation of `fn` or else `initial`)
+, and index is the index number.
+* `arg {Any-Total}` The initial value for the "running total" object.
+* Returns: `{Any-Total}` The final running total object as returned from the final invocation of the function on the last list member.
+
+
+### fluid.remove_if(source, fn[, target])
+
+Scan through a list or hash of objects, removing those which match a predicate. The source structure *** will be modified in place ***.
+
+* `source {Array|Object}` The list or hash of objects to be scanned over.
+* `fn {Function: (object {Any}, index {Number|String}) → Booleanish }` A predicate function determining whether an element should be
+removed. This accepts the standard signature `(object, index)`. If the predicate returns a "truthy" value, the corresponding object
+will be removed from the source structure.
+* `target {Array|Object}` [optional] A target object of the same type as `source`, which will
+receive any objects removed from it.
+* Returns: `{Array|Object}` Either `target`, containing the removed elements, if it was supplied, or else `source`
+modified by the operation of removing the matched elements.
+
+
+### fluid.getMembers(holder, path)
+
+Extracts a particular member from each top-level member of a container, returning a new container of the same type.
+* `holder {Array|Object}` The container to be filtered
+* `path {String|Array of String}` An [EL path](FrameworkConcepts.md#el-paths) to be fetched from each top-level member
+* Returns: `{Array|Object}` A container of the same type as `holder`, with the members of `holder` deferenced by `path`.
+
+
+### fluid.filterKeys(toFilter, keys[, exclude])
+
+Accepts an object to be filtered, and a list of keys. Either all keys not present in
+the list are removed, or only keys present in the list are returned. The suppled object *** is not modified *** but will be shallow cloned by this operation.
```

Comment:
I think the explanation of this function – the either/or section – needs some correction, because it seems to be describing the same thing twice...